static/js/place/chart.tsx

```diff
@@ -533,33 +628,83 @@ class Chart extends React.Component<ChartPropType, ChartStateType> {
       return this.state.choroplethDataGroup
         ? this.state.choroplethDataGroup.exploreUrl
         : "";
-    } else {
-      return this.props.trend
-        ? this.props.trend.exploreUrl
-        : this.props.snapshot.exploreUrl;
     }
+    if (this.props.chartType === chartTypeEnum.RANKING) {
+      return this.state.rankingChartDataGroup
+        ? this.state.rankingChartDataGroup.exploreUrl
+        : "";
+    }
+    return this.props.trend
+      ? this.props.trend.exploreUrl
+      : this.props.snapshot.exploreUrl;
   }
 
   private getSources(): string[] {
     if (this.props.chartType == chartTypeEnum.CHOROPLETH) {
       return this.state.choroplethDataGroup
         ? this.state.choroplethDataGroup.sources
         : [];
-    } else {
-      return this.props.trend
-        ? this.props.trend.sources
-        : this.props.snapshot.sources;
     }
+    if (this.props.chartType === chartTypeEnum.RANKING) {
+      return this.state.rankingChartDataGroup
+        ? this.state.rankingChartDataGroup.sources
+        : [];
+    }
+    return this.props.trend
+      ? this.props.trend.sources
+      : this.props.snapshot.sources;
   }
 
   private getDateString(): string {
     if (this.props.chartType == chartTypeEnum.CHOROPLETH) {
       return this.state.choroplethDataGroup
         ? "(" + this.state.choroplethDataGroup.date + ")"
         : "";
-    } else {
-      return this.props.snapshot ? "(" + this.props.snapshot.date + ")" : "";
     }
+    if (this.props.chartType === chartTypeEnum.RANKING) {
+      return this.state.rankingChartDataGroup
+        ? "(" + this.state.rankingChartDataGroup.date + ")"
+        : "";
+    }
+    return this.props.snapshot ? "(" + this.props.snapshot.date + ")" : "";
+  }
+
+  private getRankingChartData(
+    data: RankingChartDataGroup
+  ): { lowest: Point[]; highest: Point[] } {
+    const lowestAndHighestDataPoints = { lowest: [], highest: [] };
+    if (
+      data.numDataPoints >= MIN_RANKING_DATAPOINTS &&
+      data.numDataPoints <= MAX_RANKING_DATAPOINTS
+    ) {
+      const sliceNumber = Math.floor(data.numDataPoints / 2);
+      lowestAndHighestDataPoints.lowest = data.data
+        .slice(-sliceNumber)
+        .reverse();
+      lowestAndHighestDataPoints.highest = data.data.slice(0, sliceNumber);
+      return lowestAndHighestDataPoints;
+    }
+    const sliceNumber = Math.floor(MAX_RANKING_DATAPOINTS / 2);
+    lowestAndHighestDataPoints.lowest = data.data.slice(-sliceNumber).reverse();
+    lowestAndHighestDataPoints.highest = data.data.slice(0, sliceNumber);
+    return lowestAndHighestDataPoints;
+  }
+
+  private getRankingChartContainerTitle(): string {
+    const placeName = this.props.names[this.props.dcid] || this.props.dcid;
+    const currentPlaceRankAndValue = this.state.rankingChartDataGroup.data.find(
+      ({ placeDcid }) => placeDcid === this.props.dcid
```

Comment:
oh, have never seen this syntax before, it works?
usually, we do: `(d) => d.placeDcid === this.props.dcid`

Reply:
Yeup, Just learned from the [website ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)that destructuring can be used here